### PR TITLE
include more testing for safer loading of various yaml types

### PR DESCRIPTION
**Pull request by addyess**
[LP#2025087](https://bugs.launchpad.net/ops-lib-manifest/+bug/2025087)  More unit testing to guarantee that various yaml types are loaded correctly

### CODE REVIEW

File: ops/manifests/manifest.py [hunk 1]
```diff
@@ -208,25 +208,42 @@ def create_crd(rsc):
...
                     # found a non-dict item?  Let's log it
-                    log.warning(f"Ignoring non-dictionary resource rsc='{rsc}'")
-                elif rsc.get("kind") == "List":
+                    log.warning(
+                        f"Ignoring non-dictionary resource rsc='{rsc}' in {filepath}"
+                    )
+                elif not rsc.get("kind") or not rsc.get("apiVersion"):
+                    log.warning(
+                        f"Ignoring non-kubernetes resource rsc='{rsc}' in {filepath}"
+                    )
```
Comment (Cynerva) [inline ops/manifests/manifest.py:230]: I can imagine a worst case scenario where `rsc` is a Secret that's missing `apiVersion` or `kind`, in which case we would accidentally log secret data here.

We're not generally loading secrets via this method right? Seems like those generally fall outside the realm of ops-lib-manifest? So maybe it's not a concern worth addressing.

**Review by Cynerva — APPROVED** [review]
LGTM. I see a potential corner case worth considering but I *think* it's unlikely enough that it should be fine.

File: ops/manifests/manifest.py (hunk 1 above)
Comment (addyess) [reply ops/manifests/manifest.py:230]: I'm not sure i understand a `Secret` that doesn't have an `apiVersion` or `kind`

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: mysecret
type: Opaque
data:
  username: YWRtaW4=
  password: YmlyZHNhcmVudHJlYWw=
```

Comment (Cynerva) [reply ops/manifests/manifest.py:230]: Yeah, it'd be a really extreme corner case where somebody rendered a file that's *supposed* to be a Kubernetes secret resource, but then forgot to include the `apiVersion` or `kind` field when rendering it. Then if that flows through here, it'll get logged as an invalid resource, but the secret data would appear in the logs.

Again, seems pretty unlikely and it's probably not an issue worth addressing.

Comment (addyess) [reply ops/manifests/manifest.py:230]: right, in this case i'd really not expect this case.  usually the manfiests are flat file checked into source -- not files that require to be first rendered.

this library hasn't taken extreme care to ensure any data hasn't leaked yet -- likely should be a bug on the library i'll go add right now.

**Pull request merged**